src/details/ArborX_DetailsTreeTraversal.hpp

```diff
@@ -290,6 +291,28 @@ struct TreeTraversal<BVH, Predicates, Callback, NearestPredicateTag>
     _callback(predicate, 0);
   }
 
+  template <typename Query, typename T = Callback>
+  KOKKOS_FUNCTION
+      std::enable_if_t<std::is_same<T, Callback>{} &&
+                           !callback_exact_distance<Callback, Query>{},
+                       float>
+      exactDistance(Query const &, int, float distance) const
```

Comment:
We need to come up with a better name.

Reply:
`userDistance`, `primitiveDistance`, `fineDistance`?